plugins/git-commit/git-commit.plugin.zsh

```diff
@@ -14,6 +14,56 @@ _git_commit_aliases=(
   'wip'
 )
 
+_git_commit_flags=(
+  '-s'
+  '--scope'
+  '-a'
+  '--attention'
+  '-sa'
+  '-as'
+)
+
+remove_flags() {
+  local regex=$(IFS="|"; echo "${_git_commit_flags[*]}")
+
+  echo $(sed -E "s/\s($regex)\s|\s($regex)$//g" <<< "$1")
+};
+
+build_message () {
+  local argv=("$@")
+  local argc=${#@}
+  local i=1
+
+  while [[ $i -lt $argc ]];
+  do
+    local flag="${argv[$i]}"
+    local value="${argv[$i + 1]}"
+    local next=$[$i + 1]
+
+    if [[ "$flag" =~ (-sa|-as) ]]; then
+      local attention='!'
+      local scope="($value)"
+      i=$next
+    fi
+
+    if [[ "$flag" =~ (-a|--attention) ]]; then
+      local attention='!'
+      i=$next
+    fi
+
+    if [[ "$flag" =~ (-s|--scope) && -n "$value" ]]; then
+      local scope="($value)"
+      i=$next
+    fi
+  i=$next
+  done
+
+  local template="'$type'${scope}$attention: ${@}";
+  local message=$(remove_flags "$template" "${_git_commit_flags[@]}")
```

Comment:
Since `remove_flags` is only used once, could you embed it here?

Reply:
Very good, it makes sense!

Reply:
Adding `noglob` to avoid `*` being expanded!

Reply:
Thank you, I understood differently!
> What I mean here is to do the following directly: